tests/test_atproto_client/models/tests/test_string_formats.py

```diff
@@ -0,0 +1,198 @@
+from pathlib import Path
+from typing import List
+
+import pytest
+from atproto_client.exceptions import ModelError
+from atproto_client.models import string_formats
+from atproto_client.models.string_formats import _OPT_IN_KEY
+from atproto_client.models.utils import get_or_create
+from pydantic import BaseModel, TypeAdapter, ValidationError
+
+# Test data sourced directly from bluesky-social/atproto repo:
+# https://github.com/bluesky-social/atproto/tree/main/interop-test-files/syntax
+INTEROP_TEST_FILES_DIR: Path = Path('tests/test_atproto_client/interop-test-files/syntax')
+
+
+def get_test_cases(filename: str) -> List[str]:
+    """Get non-comment, non-empty lines from an interop test file."""
+    return [
+        line.strip()
+        for line in INTEROP_TEST_FILES_DIR.joinpath(filename).read_text().splitlines()
+        if line.strip() and not line.startswith('#')
+    ]
+
+
+@pytest.fixture
+def valid_handles() -> List[str]:
+    return get_test_cases('handle_syntax_valid.txt')
+
+
+@pytest.fixture
+def valid_dids() -> List[str]:
+    return get_test_cases('did_syntax_valid.txt')
+
+
+@pytest.fixture
+def valid_nsids() -> List[str]:
+    return get_test_cases('nsid_syntax_valid.txt')
+
+
+@pytest.fixture
+def valid_aturis() -> List[str]:
+    return get_test_cases('aturi_syntax_valid.txt')
+
+
+@pytest.fixture
+def valid_datetimes() -> List[str]:
+    return get_test_cases('datetime_syntax_valid.txt')
+
+
+@pytest.fixture
+def valid_tids() -> List[str]:
+    return get_test_cases('tid_syntax_valid.txt')
+
+
+@pytest.fixture
+def valid_record_keys() -> List[str]:
+    return get_test_cases('recordkey_syntax_valid.txt')
+
+
+@pytest.fixture
+def valid_data(
+    valid_handles: List[str],
+    valid_dids: List[str],
+    valid_nsids: List[str],
+    valid_aturis: List[str],
+    valid_datetimes: List[str],
+    valid_tids: List[str],
+    valid_record_keys: List[str],
+) -> dict:
+    return {
+        'handle': valid_handles[0],
+        'did': valid_dids[0],
+        'nsid': valid_nsids[0],
+        'at_uri': valid_aturis[0],
+        'cid': 'bafyreidfayvfuwqa2beehqn7axeeeaej5aqvaowxgwcdt2rw',  # No interop test file for CID
+        'datetime': valid_datetimes[0],
+        'tid': valid_tids[0],
+        'record_key': valid_record_keys[0],
+        'uri': 'https://example.com',  # No interop test file for URI
+        'language': 'en-US',  # No interop test file for language
+    }
+
+
+@pytest.fixture
+def invalid_data() -> dict:
+    return {
+        'handle': get_test_cases('handle_syntax_invalid.txt')[0],
```

Comment:
could we parametrize it as well? to use all the test cases from .txt files instead of the first one